packages/apps/human-app/frontend/src/modules/worker/components/sign-in/sign-in-form.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable camelcase -- ...*/
+import { FormProvider, useForm } from 'react-hook-form';
+import { Box, Grid, Typography } from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { Input } from '@/shared/components/data-entry/input';
+import { Button } from '@/shared/components/ui/button';
+import { Password } from '@/shared/components/data-entry/password/password';
+import { routerPaths } from '@/router/router-paths';
+import { FormCaptcha } from '@/shared/components/h-captcha';
+import { type SignInDto } from '@/modules/worker/services/sign-in/types';
+import { signInDtoSchema } from '@/modules/worker/services/sign-in/schema';
+import { useResetMutationErrors } from '@/shared/hooks/use-reset-mutation-errors';
+
+interface SignInFormProps {
+  onSubmit: (data: SignInDto) => void;
+  error?: unknown;
+  isLoading?: boolean;
+  resetMutation: () => void;
+}
+
+export function SignInForm({
+  onSubmit,
+  error,
+  isLoading,
+  resetMutation,
+}: SignInFormProps) {
+  const { t } = useTranslation();
+
+  const methods = useForm<SignInDto>({
+    defaultValues: {
+      email: '',
+      password: '',
+      h_captcha_token: '',
+    },
+    resolver: zodResolver(signInDtoSchema),
+  });
+
+  useResetMutationErrors(methods.watch, resetMutation);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
You probably need to create submit handler in different way:
```suggestion
  const handleSubmit = methods.handleSubmit(onSubmit);
```
Or for simplicity you can pass  `methods.handleSubmit(onSubmit)` to `onSubmit` directly

Reply:
Hey, I tried to do this (and similar things) but types here are not compatibile: `Promise-returning function provided to attribute where a void return was expected.`. 

`methods.handleSubmit` returns `Promise<void>` while form's `onSubmit` expects `FormEventHandler<T> | undefined;`


I can do something like:
```
  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    void methods.handleSubmit(onSubmit)(event);
  };

  return (
    <FormProvider {...methods}>
      <form onSubmit={handleSubmit}>
``` 

but it's not a big change, WDYT?

Reply:
@mpblocky weird peculiarity, thanks for the info

Then it's fine to keep it as is, no need to add `event.preventDefault()` because it's already handled by react-hook-form ([ref](https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L1206); otherwise we would get page reloaded as a default form submit action)